File: test_filters.py
import json

with open('public/data/GrandRift.json', 'r') as f:
    d = json.load(f)

baseEvents = []
sDate = '10 Feb 2026'
sMatch = 'all'
sUser = 'all'

for mId, m in d.items():
    if sDate != 'all' and m['date'] != sDate:
        continue
    if sMatch != 'all' and mId != sMatch:
        continue

    evs = m['events']
    if sUser != 'all':
        evs = [e for e in evs if e[0] == sUser]
    
    baseEvents.extend(evs)

print(f"Filtered: {len(baseEvents)} events matching Date: {sDate}")

File: test_filters.js
const fs = require('fs');

const d = JSON.parse(fs.readFileSync('public/data/GrandRift.json', 'utf8'));
let baseEvents = [];

let sDate = '14 Feb 2026';
let sMatch = 'all';
let sUser = 'all';

Object.keys(d).forEach(mId => {
    const m = d[mId];
    if (sDate !== 'all' && m.date !== sDate) return;
    if (sMatch !== 'all' && mId !== sMatch) return;

    let evs = m.events;
    if (sUser !== 'all') evs = evs.filter(e => e[0] === sUser);
    
    baseEvents.push(...evs);
});

console.log(`Filtered: ${baseEvents.length}`);
